core/src/main/scala/org/http4s/parser/SimpleHeaders.scala

```diff
@@ -23,11 +23,24 @@ import headers._
 import java.net.InetAddress
 import org.http4s.util.CaseInsensitiveString._
 
+import scalaz.NonEmptyList
+
 /**
  * parser rules for all headers that can be parsed with one simple rule
  */
 private[parser] trait SimpleHeaders { self: HttpParser =>
 
+  def ALLOW(value: String): ParseResult[Allow] = {
+    new Http4sHeaderParser[Allow](value) {
+      def entry = rule {
+        oneOrMore(Token).separatedBy(ListSep) ~ EOL ~>  { ts: Seq[String] =>
+          val ms = ts.map(Method.fromString(_).getOrElse(sys.error("Impossible. Please file a bug report.")))
```

Comment:
In this case it should be impossible because the header parser well fall
before trying to convert to a seq of methods.
On Mar 23, 2015 10:12 AM, "Ross A. Baker" <notifications@github.com> wrote:

> In core/src/main/scala/org/http4s/parser/SimpleHeaders.scala
> <https://github.com/http4s/http4s/pull/235#discussion_r26939011>:
>
> >  /**
> >   * parser rules for all headers that can be parsed with one simple rule
> >   */
> >  private[parser] trait SimpleHeaders { self: HttpParser =>
> >
> > +  def ALLOW(value: String): ParseResult[Allow] = {
> > +    new Http4sHeaderParser[Allow](value) {
> > +      def entry = rule {
> > +        oneOrMore(Token).separatedBy(ListSep) ~ EOL ~>  { ts: Seq[String] =>
> > +          val ms = ts.map(Method.fromString(_).getOrElse(sys.error("Impossible. Please file a bug report.")))
>
> Not impossible, in case someone tried to put a non-token character in
> there. But really, really, really weird.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/http4s/http4s/pull/235/files#r26939011>.
>